samples/boards/arc_secure_services/src/main.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2018 Synopsys.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <soc.h>
+
+#if defined(CONFIG_SOC_NSIM_SEM)
+#define NORMAL_FIRMWARE_ENTRY 0x40000
+#elif defined(CONFIG_SOC_EMSK)
+#define NORMAL_FIRMWARE_ENTRY 0x20000
+#endif
+
+
+#define STACKSIZE 1024
+#define PRIORITY 7
+#define SLEEPTIME 1000
+
+
+void threadA(void *dummy1, void *dummy2, void *dummy3)
```

Comment:
is a thread required in Secure mode? We don't support mixed secure/normal threads (yet...), the Secure part is more like a library and initial entry point before handing over to Normal mode running full Zephyr.

Reply:
It's a complicate case. It thread is not required, just disable the multithreading in Zephry. If we can let two zephyr run together, it will be simple for baremetal case.